### PR TITLE
Functionality to copy information from the about popup

### LINKED ISSUE #1493: Improve user issue reporting - add copy feature to About popup
When reporting issues, it is useful to know the current version and platform information. This is already provided in the About popup, so it would be convenient to add a feature to copy that information to the clipboard for use in bug (or feature) reports.

The code already wraps copy functionality, so the primary aspect here is to construct the information appropriately from the data available in the About popup.

We expect to standardize on <kbd>c</kbd> for copying, though the command string is somewhat in flux at present. See for example `COPY_STREAM_EMAIL`.

This was previously discussed in `#zulip-terminal>View object has no attribute topic_w` on chat.zulip.org.

### PR DESCRIPTION
**Pull request by rsashank**
### What does this PR do, and why?
Adds hotkey 'c' to copy information from the about popup. It's copied in this format:

```
#### Application
Zulip Terminal: 0.7.0+git

#### Server
Version: 9.0-dev-3226-gbc50a9743b
Feature level: 265

#### Application Configuration
Theme: zt_dark
Autohide: disabled
Maximum footlinks: 3
Color depth: 256
Notifications: disabled
Exit confirmation: enabled

#### Detected Environment
Platform: WSL
Python: 3.10.12 (CPython) 
```

### External discussion & connections

- [x] Discussed in **#zulip-terminal** in `Copy Information from AboutPopup #T1493`
- [x] Fully fixes #1493
- [ ] Partially fixes issue #
- [ ] Builds upon previous unmerged work in PR #
- [ ] Is a follow-up to work in PR #
- [ ] Requires merge of PR #
- [ ] Merge will enable work on #

### How did you test this?

- [x] Manually - Behavioral changes
- [x] Manually - Visual changes
- [x] Adapting existing automated tests
- [x] Adding automated tests for new behavior (or missing tests)
- [ ] Existing automated tests should already cover this (*only a refactor of tested code*)

### Self-review checklist for each commit
- [x] It is a [minimal coherent idea](https://github.com/zulip/zulip-terminal#structuring-commits---speeding-up-reviews-merging--development)
- [x] It has a commit summary following the [documented style](https://github.com/zulip/zulip-terminal#structuring-commits---speeding-up-reviews-merging--development) (title & body)
- [ ] It has a commit summary describing the  motivation and reasoning for the change
- [x] It individually passes linting and tests
- [x] It contains test additions for any new behavior
- [x] It flows clearly from a previous branch commit, and/or prepares for the next commit

### Visual changes    

| Before | After |
|--------|--------|
| image| image|

### CODE REVIEW

**Review by Niloth-p — COMMENTED** [review]
Good PR, Sashank! This should really help issue reporters.
It works fine.

Clarifying what 'information' would help not only in the help texts, but also in the commit messages / descriptions.

Hard coding the string does not seem sustainable. It's preferable to just generate the string like this.
```python
sections = []
for section_title, properties in contents:
	formatted_properties = '\n'.join(f"{prop}: {value}" for prop, value in properties)
	sections.append(f"{section_title}\n{formatted_properties}")
self.copy_info = '\n\n'.join(sections)

about_keys = ", ".join(map(repr, display_keys_for_command("COPY_ABOUT_INFO")))
contents.append((" ", [(f"Press {about_keys} to copy this information.", "")]))
```
Add the keys to the contents after extracting the string version.

I'm wondering if we're going to be adding enough copy commands to create a help category just for them.

**Comment by neiljp** [comment]
@rsashank Some quick notes based on the content of the PR right now:
- Re an 'empty category', I was thinking more along the lines of a category having no contents, but the category title being the copy hint, making the text stand out like the other titles
- To style the help key similar to elsewhere in the UI, we want it to have `[]` around the key, and be to the right of the text
- Re the copied content, having it suitable for pasting into an issue would be preferable, so using some (sub)title markdown for each category title would look good, and also help them stand out in plain text form
- I support a flexible implementation closer to @Niloth-p's suggestion re the formatting, since the contents of the popup may change over time and a more algorithmic approach will automatically stay up to date (though the tests should be more static, to track behavioral changes)

File: docs/hotkeys.md [hunk 1]
```diff
@@ -10,6 +10,7 @@
 |Show/hide about menu|<kbd>Meta</kbd> + <kbd>?</kbd>|
 |Go Back|<kbd>Esc</kbd>|
 |Open draft message saved in this session|<kbd>d</kbd>|
+|Copy information from the About popup|<kbd>c</kbd>|
```
[the remark this answers is not in the document]
Comment (rsashank) [reply docs/hotkeys.md]: application information?

**Comment by Niloth-p** [comment]
This lgtm!
The feedback has been integrated into the code.
And the PR description comment is quite helpful and accurate.

File: docs/hotkeys.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (neiljp) [reply docs/hotkeys.md]: I adjusted this to include About *Menu*, and specifically to the clipboard; it's also not just regarding the application, so it seems fine to be this broad.

File: tests/ui_tools/test_popups.py [hunk 2]
```diff
@@ -261,7 +261,8 @@ def test_categories(self) -> None:
             "Application",
             "Server",
             "Application Configuration",
-            "Detected environment",
+            "Detected Environment",
```
[the remark this answers is not in the document]
Comment (neiljp) [reply tests/ui_tools/test_popups.py:278]: I split this out in my push just now - it's generally just easier to track what's happening in the git log if each commit makes a single change.

**Comment by neiljp** [comment]
@rsashank Thanks for this update :+1:

Since this looked very close and you're due to be away for a few days, this seemed close to merge with some minor changes - and opens up editing the issue templates to reference this feature!

I made some minor text updates, and kept the change of case in an earlier commit to keep it clearly separate. I realized before pushing back here that you didn't have a test for the copied text output, so added a commit with that myself.

**Pull request merged**